File: apps/ios/TaakHelden/Core/DesignSystem/AvatarCatalog.swift
import Foundation

enum AvatarCatalog {
    private static let emojiByID: [String: String] = [
        "fox": "🦊",
        "panda": "🐼",
        "lion": "🦁",
        "octopus": "🐙",
        "unicorn": "🦄",
        "tiger": "🐯",
    ]

    static let selectableIDs = ["fox", "panda", "lion", "octopus", "unicorn", "tiger"]

    static func id(forEmoji emoji: String) -> String {
        emojiByID.first(where: { $0.value == emoji })?.key ?? selectableIDs[0]
    }

    static func emoji(for avatarID: String?) -> String {
        guard let avatarID, let emoji = emojiByID[avatarID] else {
            return "🦊"
        }
        return emoji
    }

    static func ageBand(from ageMode: String?) -> ChildAgeBand {
        switch ageMode {
        case "young": return .young
        case "teen": return .teen
        default: return .mid
        }
    }

    static func ageBand(fromBirthYear birthYear: Int) -> ChildAgeBand {
        let age = Calendar.current.component(.year, from: Date()) - birthYear
        if age <= 7 { return .young }
        if age >= 13 { return .teen }
        return .mid
    }
}

File: apps/android/core/src/main/kotlin/nl/taakhelden/core/designsystem/AvatarCatalog.kt
package nl.taakhelden.core.designsystem

import nl.taakhelden.core.auth.ChildAgeBand
import java.time.Year

public object AvatarCatalog {
    private val emojiById: Map<String, String> = mapOf(
        "fox" to "🦊",
        "panda" to "🐼",
        "lion" to "🦁",
        "octopus" to "🐙",
        "unicorn" to "🦄",
        "tiger" to "🐯",
    )

    public val selectableIds: List<String> =
        listOf("fox", "panda", "lion", "octopus", "unicorn", "tiger")

    public val selectableEmojis: List<String> = selectableIds.map { emojiById.getValue(it) }

    public fun idForEmoji(emoji: String): String =
        emojiById.entries.firstOrNull { it.value == emoji }?.key ?: selectableIds[0]

    public fun emojiFor(avatarId: String?): String =
        avatarId?.let { emojiById[it] } ?: "🦊"

    public fun ageBandFrom(ageMode: String?): ChildAgeBand = when (ageMode) {
        "young" -> ChildAgeBand.YOUNG
        "teen" -> ChildAgeBand.TEEN
        else -> ChildAgeBand.MID
    }

    public fun ageBandFromBirthYear(
        birthYear: Int,
        currentYear: Int = Year.now().value,
    ): ChildAgeBand {
        val age = currentYear - birthYear
        return when {
            age <= 7 -> ChildAgeBand.YOUNG
            age >= 13 -> ChildAgeBand.TEEN
            else -> ChildAgeBand.MID
        }
    }
}
